### PR TITLE
arch: Rename up_[early]serialinit to [arm64|riscv|x86_64][early]serialinit

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

The naming standard at:
https://cwiki.apache.org/confluence/display/NUTTX/Naming+of+Architecture%2C+MCU%2C+and+Board+Interfaces
requires that all MCU-private function begin with the name of the architecture, not up_.

follow the change from: https://github.com/apache/nuttx/pull/930
## Impact

arm64, riscv-v and x86[_64]

## Testing

CI

### CODE REVIEW

**Comment by acassis** [comment]
@xiaoxiang781216 please consider including in the log message the motivation to change the function name, as people suggested in the mailing list. I think a good benefit of getting rid of these tons of up_* inside all arch is it make IDE symbols search easier. Actually almost none IDE is able to resolve NuttX symbol because it uses same symbol name in different arch.

**Comment by jerpelea** [comment]
please add a proper commit message to justify the change

**Comment by xiaoxiang781216** [comment]
Done, @jerpelea @acassis

File: arch/x86/src/qemu/qemu_serial.c
Comment (pkarashchenko) [inline arch/x86/src/qemu/qemu_serial.c:103]: Please add newline at the end of file

File: arch/arm64/src/fvp-v8r/fvp_lowputc.S
Comment (pkarashchenko) [inline arch/arm64/src/fvp-v8r/fvp_lowputc.S:59]: I do not fully understand why this is removed and not renamed. Could you please make a short explanation here?

Comment (xiaoxiang781216) [reply arch/arm64/src/fvp-v8r/fvp_lowputc.S:59]: https://github.com/apache/nuttx/pull/8716 already add arm64_earlyserialinit but forget to remove up_earlyserialinit.

File: arch/x86/src/qemu/qemu_serial.c
Comment (xiaoxiang781216) [reply arch/x86/src/qemu/qemu_serial.c:103]: Done.

File: arch/arm64/src/fvp-v8r/fvp_lowputc.S
Comment (pkarashchenko) [reply arch/arm64/src/fvp-v8r/fvp_lowputc.S:59]: But the things done here are completely different from what is done in `arm64_earlyserialinit`. This code is branched very early from ASM code and uses
```
GTEXT(arm64_lowputc)
SECTION_FUNC(text, arm64_lowputc)
    ldr   x15, =CONFIG_UART0_BASE
    early_uart_ready x15, w2
    early_uart_transmit x15, w0
    ret
```
for the low level output. If we remove the `up_earlyserialinit` then non of the lines like `PRINT(primary_boot, "- Ready to Boot Primary CPU\r\n")` will be ever printed.
I will take a look into https://github.com/apache/nuttx/pull/8716 , but seems like we might be breaking an important path of debugging early startup process

Comment (pkarashchenko) [reply arch/arm64/src/fvp-v8r/fvp_lowputc.S:59]: I see many mess in the code like in `arm_nommuhead.S` that calls `showprogress` -> `arm_lowputc` before
```
#ifdef USE_EARLYSERIALINIT
	bl		arm_earlyserialinit
#endif
```
And that is fine if underlying BL did all the work to mux the pins and setup the UART registers, but most probably will be unless in other case.
Some of the MCUs have `xxx_lowsetup()` that is similar to what `up_earlyserialinit` is doing here.
I think that we should hold off a bit here as this is not just removing of something unused and re-inspect how the things should work

Comment (xiaoxiang781216) [reply arch/arm64/src/fvp-v8r/fvp_lowputc.S:59]: @qinwei2004 will provide the patch to correct the early print support after this get merged.

Comment (qinwei2004) [reply arch/arm64/src/fvp-v8r/fvp_lowputc.S:59]: @acassis @jerpelea @pkarashchenko 
it need to do early initialize when we want to print something in arm64_head.S
I cannot call arm64_earlyserialinit or other C function because it's very early (before BSS initialize)
and with a high ELs
At next patch, I will add arm64_earlyprintinit which is only initialize serial just setting BAUDRATE to bring serial into base work state in assembly language. A complete initiation will be done at arm64_earlyserialinit

I consider someone can also logging the booting message through rewriting arm64_lowputc with other debug method (eg semihosting, ARM debug channel etc), not only serial

but I check the function again and happy to find we don't need initialize it anymore. new version qemu or new option has do it for us. so I remove this code

**Comment by xiaoxiang781216** [comment]
@acassis @jerpelea @pkarashchenko do you have more comment?

File: arch/arm64/src/fvp-v8r/fvp_lowputc.S
Comment (pkarashchenko) [reply arch/arm64/src/fvp-v8r/fvp_lowputc.S:59]: But in `bl arm64_earlyserialinit` you are doing exactly what you say you can't do "I cannot call arm64_earlyserialinit or other C function because".

**Comment by pkarashchenko** [comment]
> @acassis @jerpelea @pkarashchenko do you have more comment?

Yes. I continue discussion. I think that current change in raw and I do not fully understand the final behaviour. Now we have two functionalities:
1. Possibility to do very early print from assembly files that can be used even before `.data` and `.bss` init. This use some stack-less initialisations and printing
2. `ARCH_earlyserialinit` that is called from C code during boot process.
I feel that this PR is mixing things together.

**Comment by xiaoxiang781216** [comment]
> > @acassis @jerpelea @pkarashchenko do you have more comment?
> 
> Yes. I continue discussion. I think that current change in raw and I do not fully understand the final behaviour. Now we have two functionalities:
> 
> 1. Possibility to do very early print from assembly files that can be used even before `.data` and `.bss` init. This use some stack-less initialisations and printing

To do this, a new assembly function will be added. The implementation will post after this patch is merged.

> 2. `ARCH_earlyserialinit` that is called from C code during boot process.
>    I feel that this PR is mixing things together.

ARCH_earlyserialinit is designed to callable in C before OS(mm/sched) fully initialize. arm64 is wrong to assume that ARCH_earlyserialinit can be called in assembly boot code, which will be fixed in the upcoming patch.

**Comment by pkarashchenko** [comment]
But this PR is calling `arm64_earlyserialinit` from early assembly. So do you propose to merge PR that "breaks the things" and then merge another PR that "fix the things"? What is the point in this instead of doing this properly in 1 PR?

**Comment by xiaoxiang781216** [comment]
Yes, since the print in assembly is disabled by default, the temp break isn't a big problem.

**Comment by pkarashchenko** [comment]
Currently the `arm64_lowputc` is implemented as
```
GTEXT(arm64_lowputc)
SECTION_FUNC(text, arm64_lowputc)
    ldr   x15, =UART0_BASE_ADDRESS
    early_uart_ready x15, w2
    early_uart_transmit x15, w0
    ret
```
but in C code I see
```
#if defined(CONFIG_UART1_SERIAL_CONSOLE)
#  define CONSOLE_DEV     g_uart1port         /* UART1 is console */
#  define TTYS0_DEV       g_uart1port         /* UART1 is ttyS0 */
#  define UART1_ASSIGNED  1
#endif
```
So UART0 may be even not a console UART.
And `UART0_BASE_ADDRESS` is
```
 /* Allwinner A64 UART0 Base Address */

 #define UART0_BASE_ADDRESS 0x1C28000
```
So what if MCU that is running is not `Allwinner A64`?
I agree that things are broken, but just do not want to go into "even more broken" state

**Comment by pkarashchenko** [comment]
@xiaoxiang781216  So the whole idea of this PR is just to remove `up_earlyserialinit`?

**Comment by xiaoxiang781216** [comment]
not remove, rename up_earlyserialinit to arm64_earlyserialinit, which just follow other arch convention. and xxx_earlyserialinit shouldn't be called by boot assembly code.

**Comment by qinwei2004** [comment]
@pkarashchenko ，My new patch must wait for this patch to be merged.
at new patch, I will call assembly code directly

File: include/nuttx/serial/uart_16550.h
Comment (pkarashchenko) [inline include/nuttx/serial/uart_16550.h:334]: Please add description comment

**Comment by pkarashchenko** [comment]
Ok. Let's move on with this change if the new PR is ready. I do not want to block the process despite I do not have a good explanation why this can't be done in a single PR.

**Comment by xiaoxiang781216** [comment]
> Ok. Let's move on with this change if the new PR is ready. I do not want to block the process despite I do not have a good explanation why this can't be done in a single PR.

@qinwei2004 doesn't have right to modify my PR I think.

**Comment by pkarashchenko** [comment]
> > Ok. Let's move on with this change if the new PR is ready. I do not want to block the process despite I do not have a good explanation why this can't be done in a single PR.
> 
> @qinwei2004 doesn't have right to modify my PR I think.

I think since your fork is a public repo he should be able to (unless you added restrictions there). But from the "owner" perspective I agree that it is not welcomed.
Ok. Let's move on

**Comment by xiaoxiang781216** [comment]
> > > Ok. Let's move on with this change if the new PR is ready. I do not want to block the process despite I do not have a good explanation why this can't be done in a single PR.
> > 
> > 
> > @qinwei2004 doesn't have right to modify my PR I think.
> 
> I think since your fork is a public repo he should be able to (unless you added restrictions there). But from the "owner" perspective I agree that it is not welcomed. Ok. Let's move on

by default, only commiter and PMC can modify other people's patch.

**Comment by pkarashchenko** [comment]
> > > > Ok. Let's move on with this change if the new PR is ready. I do not want to block the process despite I do not have a good explanation why this can't be done in a single PR.
> > > 
> > > 
> > > @qinwei2004 doesn't have right to modify my PR I think.
> > 
> > I think since your fork is a public repo he should be able to (unless you added restrictions there). But from the "owner" perspective I agree that it is not welcomed. Ok. Let's move on
> 
> by default, only commiter and PMC can modify other people's patch. Otherwise, the hacker can easily inject the malicious code.

I see. I recall that I was able to push changes to your repo, so didn't know about those restrictions. Thank you for educating me.

File: include/nuttx/serial/uart_16550.h
Comment (xiaoxiang781216) [reply include/nuttx/serial/uart_16550.h]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Comment by qinwei2004** [comment]
@xiaoxiang781216 @jerpelea @acassis 
So any change change is needed?? or we can merge this PR
I am waiting for merge my new change

**Pull request merged**
